### PR TITLE
Checks the correct capability when editing each meta type

### PR DESCRIPTION
**Pull request by leogermani**
This PR fixes a bug where a non-admin was not able to set the primary brand for a post.

See 1200550061930446-as-1205886576266648

It also improves the metadata registration adding the appropriate auth callback for each type.

## How to test

First confirm the bug:

* Make sure you have at least two brands
* As a non-admin, create a post
* In the editor, add the post to 2 or more brands
* A "Primary brand" selector should appear
* Select a primary brand and try to save the post
* Notice you will get a permission error

Checkout this branch

* Repeat the steps and confirm you are now able to set the primary brand

### CODE REVIEW

File: includes/class-meta.php [hunk 1]
```diff
@@ -33,16 +33,15 @@ public static function init() {
 	 */
 	public static function register_option() {
 		$type   = static::get_schema()['type'] ?? 'string';
+		$cap    = static::get_capability();
```
Comment (miguelpeixe) [inline includes/class-meta.php:36]: Isn't this line superfluous? `$cap` is never used here.

File: includes/class-meta.php [hunk 2]
```diff
@@ -86,6 +85,39 @@ public static function get_post_types() {
...
+		if ( 'post' === static::$type ) {
+			return 'edit_post'; // singular, to check meta cap against the post id.
+		}
+		if ( 'term' === static::$type ) {
+			$tax_object = get_taxonomy( static::get_taxonomy() );
+			return $tax_object->cap->manage_terms;
+		}
+		if ( 'user' === static::$type ) {
+			return 'edit_users';
+		}
```
Comment (miguelpeixe) [inline includes/class-meta.php:115]: Neat solution for this abstraction!

**Review by miguelpeixe — CHANGES_REQUESTED** [review]
Working as described! Just one question:

File: includes/class-meta.php (hunk 1 above)
Comment (leogermani) [reply includes/class-meta.php]: yes! left over from a previous interaction. thanks for spotting

**Pull request merged**